### ISSUE TITLE
Scrapy script works only for some parts on a webpage

### ISSUE THREAD

**Issue by madhurtewani-wow** [scrapy/scrapy#3188]
I'm trying to scrape data from https://www.abcxyz.com/search/category-speaker/.

I'm able to scrape data from some elements of the page using both selector.xpath and response.css.
But not able to scrap data from some other elements of the same page using both selector.xpath and response.css.

My script is as follows and I have put comment for the lines that are working and not working.
```
import scrapy
from scrapy import Selector

class AbcxyzSpider(scrapy.Spider):
	name = 'abcxyzspider'
	ROBOTSTXT_OBEY = False
	rotate_user_agent = True
	rotate_proxies = True
	start_urls = ['https://www.abcxyz.com/search/category-speaker/']

	def parse(self, response):
		print response
		sel = Selector(response)
		print sel.xpath('//span[@class="order-count"]/text()').extract_first() #working
		print sel.xpath('//span[@class="products__item-price--final"]/span/text()').extract_first() #not_working

		print response.css('span.order-count::text').extract_first() #working
		print response.css('span.products__item-price--final > span::text').extract_first() #not_working
```

Please help.

**Comment by colinfike** [comment]
It seems to me that digikala is asynchronously loading in the products that you are trying to scrape. What is probably happening is scrapy is requesting the page without rendering/executing the javascript to load the products so you are trying to scrape a page without what you want.

I'd look in to [Splash](https://github.com/scrapinghub/splash) and how/if you can use it with scrapy. I saw a few results when I googled for using scrapy and splash so I would start there.

**Comment by leonardfrank** [comment]
You need to check out if the data you want to fetch exists in the source code. If not, use the developer tool to capture the source file where the data is in.If you find that file, fake the request and get what you want.

**Comment by madhurtewani-wow** [comment]
@colinfike and @leonardfrank thank you very much for the help.

@leonardfrank, I have tried what you have mentioned and it did work for me partially.
But then on a different page of same website, I am not able to find the corresponding request from Chrome DevTools (FYI not using it for the first time) for the data I want to scrap.

I tried couple of other things but nothing worked. I have managed to work without the data from the different page for now.
